src/kernels/levelx/xconvgemm_part1.opencl

```diff
@@ -82,10 +87,17 @@ INLINE_FUNC real GlobalToLocalCheckedImage(const __global realMD* restrict image
       int idm = mg + GetGroupID0()*WGD;
       int idk = kg + kwg;
 
+      const int w_id = idm % output_w;
+      const int h_id = idm / output_w;
```

Comment:
So this (and similar changes elsewhere) were the bugs in the kernel? Because I had everything implemented, but things didn't work well when running the tests & the tuner, so I suspected a couple of things were wrong.

Reply:
Yes, it was a bug. `w_id` and `h_id` are also calculated in `Xconvgemm` function, but these two variables are skipped using `MWID`.